api/routers/auth.js

```diff
@@ -25,11 +27,34 @@ function onSuccess(req, res) {
 
 router.get('/logout', passport.logout(idp));
 router.get('/login', redirectIfAuthenticated, passport.authenticate(idp));
-
 router.get('/auth/github/callback', passport.authenticate('github', opts), onSuccess);
 
 // Callbacks need to be registered with CF UAA service
 router.get('/auth/uaa/callback', passport.authenticate('uaa', opts), onSuccess);
 router.get('/auth/uaa/logout', (_req, res) => res.redirect('/'));
 
+// New Github authorization only routes
+const onGithubSuccess = async (req, res) => {
+  const githubUser = req.user;
+
+  const user = await User.findByPk(req.session.passport.user);
+  await user.update({
+    ...githubUser,
+    signedInAt: new Date(),
+  });
+
+  EventCreator.audit(Event.labels.AUTHENTICATION, user, 'GitHub authentication');
+
+  const script = `
+    <script>
+      window.opener.postMessage("success", "*")
+    </script>
+  `;
+
+  res.send(script);
+};
+
+router.get('/auth/github2', passport.authenticate('github2'));
```

Comment:
Is this just going to be a transitional route until migrate completely to uaa?

Reply:
We can rename it? We still have to auth.

Reply:
Once we move away from app authentication using github would we transition the `/auth/github2` route to be `/auth/github` to authenticate the source code provider? What are the semantics to authenticate you source code provider in the future (ie`/auth/github2` vs `/auth/github` vs `/auth/provider/github`)?